collect_app/src/main/java/org/odk/collect/android/widgets/BarcodeWidget.java

```diff
@@ -113,19 +107,18 @@ protected static String stripInvalidCharacters(String data) {
 
```

Comment:
You could improve `stripInvalidCharacters()`:
1. you can have one return using ternary operator
2. There is no need to use protected static here.